protocol/src/main/java/net/md_5/bungee/protocol/Protocol.java

```diff
@@ -484,55 +526,47 @@ public final DefinedPacket createPacket(int id, int version)
             }
             if ( id > MAX_PACKET_ID )
             {
-                throw new BadPacketException( "Packet with id " + id + " outside of range " );
+                throw new BadPacketException( "Packet with id " + id + " outside of range" );
             }
 
-            Constructor<? extends DefinedPacket> constructor = protocolData.packetConstructors[id];
+            Supplier<? extends DefinedPacket> constructor = protocolData.packetConstructors[id];
             try
             {
-                return ( constructor == null ) ? null : constructor.newInstance();
-            } catch ( ReflectiveOperationException ex )
+                return ( constructor == null ) ? null : constructor.get();
+            } catch ( Throwable t )
```

Comment:
Surely try/catch unnecessary now?

Reply:
In theory some constructor could throw an exception, in that case we wrap it.

In practice we control the constructors ourself and we can safely assume this doesn't happen. I'll remove the try-catch.

Reply:
Yeah but it will just throw the exception anyway then. I suppose wrapped you get the packet ID, but that's not even necessarily helpful. I'd remove it

Reply:
Removed